rpmio/rpmpgp.c

```diff
@@ -1238,7 +1261,27 @@ int pgpPrtParamsSubkeys(const uint8_t *pkts, size_t pktlen,
 		pgpDigParamsFree(digps[count]);
 		continue;
 	    }
-	    count++;
+
+	    pgpDigParams subkey_sig = NULL;
+	    if (decodePkt(p, pend - p, &pkt) ||
+	        parseSubkeySig(&pkt, 0, &subkey_sig))
+	    {
+		pgpDigParamsFree(digps[count]);
+		break;
+	    }
+
+	    if (subkey_sig->sigtype != PGPSIGTYPE_SUBKEY_BINDING ||
+		!((subkey_sig->saved & PGPDIG_SIG_HAS_KEY_FLAGS) &&
+		  (subkey_sig->key_flags & 0x02))) {
+		/* Subkey is revoked or incapable of signing */
```

Comment:
One space length is not much of a distinction from the condition, to old tired eyes at least :monocle_face: 
When an if condition spreads over multiple lines, it usually becomes far more readable by simply adding an appropriately named helper variable or two, eg.

```
/* Is subkey revoked or incapable of signing? */
int ignore = subkey_sig->sigtype != PGPSIGTYPE_SUBKEY_BINDING ||
             !((subkey_sig->saved & PGPDIG_SIG_HAS_KEY_FLAGS) &&
               (subkey_sig->key_flags & 0x02));
if (ignore) {
    pgpDigParamsFree(digps[count]);
} else {
    ...
}
```

Reply:
Good point, thanks!